src/Admin/VMware.ScriptRuntimeService.AdminEngine/VCRegistration/VCTrustedCertificatesCollector.cs

```diff
@@ -91,60 +79,89 @@ public class VCTrustedCertificatesCollector {
          }
       }
 
+      public async Task<IEnumerable<string>> GetEncodedTrustedCertificatesAsync() {
+         _logger.LogInformation(
+                  string.Format(Resources.PerofomingOperation, Resources.EncodedTrustedCertificatesRetrieval));
+
+         using (var client = GetHttpClient()) {
+            await AuthenticateClientAsync(client);
+
+            List<string> certs = new List<string>();
+            foreach (var chain in await GetTrustedRootChainsChainsAsync(client)) {
+               foreach (var cert in await GetTrustedCertificateAsync(client, chain)) {
+                  certs.Add(cert);
+
+                  _logger.LogDebug($"Trusted CA cert: {cert}");
+               }
+            }
+
+            return certs;
+         }
+      }
+
       private async Task<IEnumerable<string>> GetTrustedCertificateAsync(HttpClient client, string chain) {
-         // https://{api_host}/api/vcenter/certificate-management/vcenter/trusted-root-chains/{chain}
-         using (var response = await client.GetAsync($"/api/vcenter/certificate-management/vcenter/trusted-root-chains/{chain}")) {
+         using (var response = await client.GetAsync($"/api/vcenter/certificate-management/vcenter/trusted-root-chains/{Uri.EscapeDataString(chain)}")) {
             response.EnsureSuccessStatusCode();
 
             var responseStr = await response.Content.ReadAsStringAsync();
             dynamic json = JsonConvert.DeserializeObject<dynamic>(responseStr);
 
             List<string> result = new List<string>();
 
-            string cert_chain = json.cert_chain.cert_chain[0];
-            int startIndex = 0;
-            var beginIndex = cert_chain.IndexOf("-----BEGIN CERTIFICATE-----", startIndex);
-            var endIndex = cert_chain.IndexOf("-----END CERTIFICATE-----", startIndex);
-            while (beginIndex > -1 && endIndex > beginIndex) {
-               result.Add(cert_chain.Substring(beginIndex, endIndex + beginIndex + 25).Trim());
+            if (json?.cert_chain?.cert_chain != null &&
+               json.cert_chain.cert_chain is IEnumerable &&
+               ((IEnumerable) json.cert_chain.cert_chain).Cast<dynamic>().Any()) {
 
-               startIndex = endIndex + 1;
-               beginIndex = cert_chain.IndexOf("-----BEGIN CERTIFICATE-----", startIndex);
-               endIndex = cert_chain.IndexOf("-----END CERTIFICATE-----", startIndex);
-            }
+               string cert_chain = json.cert_chain.cert_chain[0];
+               int startIndex = 0;
 
-            return result;
-         }
+               // Algorithm description:
+               // We get the the index of ------END CERTIFICATE----- and then
+               // the last index of -----BEGIN CERTIFICATE----- smaller than end index.
+               // This way we handle the case where we receive an invalid formated PEM
+               // in the format of *-BEGIN- -BEGIN- -END-*
```

Comment:
Why do you need to support the BEGIN, BEGIN, END case and not just produce an error?

Reply:
Well, I do support a few other cases. My reasoning is as follows:
* The server may return more than one trust chain. At least the API supports such use-case
* One of the chains may be broken
* I don't want to break SRS installation because of a faulty chain. I understand that this may bite me down the line, but having seen environments with many left-over chains, I assume a faulty chain may be just a left-over/unused one.